### ISSUE TITLE
Extra functionality for supervised_policy_trainer

### ISSUE THREAD

**Issue by wrongu** [Rochester-NRT/AlphaGo#65]
- [ ] more control over model parameters (also `input_dim` does not need to be set by the command line)
- [x] load model and weights from files to resume training from some epoch
- [x] save model json in same directory as weights
- [ ] make sure validation stats are working
- [x] write training and validation accuracy to a file for each epoch

...and maybe
- [ ] have access to `game_converter` and do conversions as-needed. It may make sense to have `game_converter` be told which files to process in what order. A multi-threaded training script may have the training thread choosing training data and adding SGF files to a preprocessing queue while the converter thread does the conversions as they are requested
